bazel/clang_tidy/plugins/BUILD

```diff
@@ -0,0 +1,6 @@
+load("//bazel:build.bzl", "redpanda_clang_tidy_plugin")
+
+redpanda_clang_tidy_plugin(
+    name = "example_check",
+    srcs = ["example_check.cc"],
```

Comment:
I think we should just put everything into a single rule here, so let's just name this "plugins" as a target name.

Reply:
yeah, one central module like e.g. https://github.com/redpanda-data/llvm-project/blob/main/clang-tools-extra/clang-tidy/cppcoreguidelines/CppCoreGuidelinesTidyModule.cpp